src/Data/Formatter/Number.purs

```diff
@@ -113,16 +113,19 @@ format (Formatter f) num =
          zeros = f.before - tens - one
          integer = Int.floor absed
          leftover = absed - Int.toNumber integer
-         rounded = Int.round $ leftover * (Math.pow 10.0 (Int.toNumber f.after))
+         factor = Math.pow 10.0 (Int.toNumber f.after)
+         rounded = Int.round $ leftover * factor
+         overflow = Int.toNumber rounded == factor

```

Comment:
Frankly, I think it would be a better solution to round it to the correct digit ahead of time, like
```PureScript
rounded = Math.round (absed * factor) / factor
integer = Int.floor rounded
leftoverDecimal = rounded - Int.toNumber integer
leftover = Int.round $ leftoverDecimal * factor
...
```
because that simplifies the logic quite a bit (you don't have to check for overflow anymore, and seems to just have much fewer assumptions overall).  I'm not sure what that means for performance though.  Would that implementation add too much runtime?

Reply:
I’m not sure — typically we use the microbench package to do little side by side tests for this. It essentially comes down to throwing a

    bench \_ -> myFunction

in front of both implementations. Would you be open to checking out the performance difference? If it’s not significantly worse then I’m happy to go for a simpler option.

Reply:
Seems to be in the noise to me:
![image](https://user-images.githubusercontent.com/15835006/90318727-6d010980-df00-11ea-9e39-1b615e2bafcb.png)
Impl 1 is my preferred implementation there where we round to the correct digit as the first step, and Impl 2 is the implementation shown where we detect overflow and recover.
These numbers are measuring the mean time over 1M trials (I had to go up to 1M to see the results between runs stay pretty consistent).  I'm guessing mean time is the measure of value here?  (Other metrics given by minibench include stddev, min, & max).  

One thing that was concerning to me is that as I increased the number of trials, the mean dropped significantly.  Most of these measures were in the single-digit microseconds, while I would often see numbers like 30 microseconds for the mean when only doing the default 1k trials.  The max values are typically up in the 2-digit milliseconds.  I wonder if the first invocation is significantly slower? Here's the same runs, but showing the max of the 1M runs instead of the mean:
![image](https://user-images.githubusercontent.com/15835006/90318843-47283480-df01-11ea-9d86-2d13ece0b274.png)

Reply:
Oh, and let me know if there's any other inputs or formats that I should benchmark.

Reply:
Nothing is coming to mind. This looks like a wash, in which case I'd rather go for the understandable version :) If you don't mind making that change I'd be happy to get it in! Thanks for benchmarking!